src/core/box_reader_impl.h

```diff
@@ -1,6 +1,7 @@
 #pragma once
 
 #include <cstring> // strcmp
+// #include <stdexcept>
```

Comment:
try/catch are part of the language. std::runtime_error requires this include. So you can remove.